pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -504,8 +504,24 @@ def _get_metric_interpolated_filepath_name(
     ) -> str:
         filepath = self.format_checkpoint_name(epoch, step, ckpt_name_metrics)
 
-        version_cnt = 0
-        while self._fs.exists(filepath) and filepath != del_filepath:
+        version_cnt = 1
+        old_ckpt_ver_0 = self.format_checkpoint_name(epoch, step, ckpt_name_metrics, ver=0)
```

Comment:
does this work if the checkpoint name contains a quantity (e.g. val loss)? wouldn't `ckpt_name_metrics` contain different values each time?